Frameworks/UIKit/UIViewController.mm

```diff
@@ -1336,28 +1349,23 @@ - (void)_addToTop:(NSNumber*)animatedValue {
 
     if ([self parentViewController] != nil) {
         if ([self modalPresentationStyle] == UIModalPresentationPopover) {
-            if (priv->_presentationController != nil) {
-                [priv->_presentationController release];
-            }
-
-            priv->_presentationController = priv->_popoverPresentationController;
-
             [priv->_popoverPresentationController _prepareForPresentation];
 
             displayPopover = ![self _hidesParent];
 
             if (displayPopover) {
-                priv->_popoverPresentationController->_managesPresentation = TRUE;
+                priv->_popoverPresentationController->_managesPresentation = YES;
```

Comment:
Thanks for simplifying this; it seems quite a bit cleaner.  Can you elaborate on why we need to support non-tablet scenarios in WinObjC though?  Would this be a spot where we could simplify the implementation by always showing popups (even on phone)?

---
In reply to: [90111314](https://github.com/Microsoft/WinObjC/pull/1464#discussion_r90111314) [](ancestors = 90111314)